src/library/dlgtagfetcher.cpp

```diff
@@ -402,4 +521,162 @@ void DlgTagFetcher::tagSelected() {
+                QPixmap(CoverArtUtils::defaultCoverLocation()));
+    }
+
+    const mixxx::musicbrainz::TrackRelease& trackRelease = m_data.m_tags[tagIndex];
+    QUuid selectedTagAlbumId = trackRelease.albumReleaseId;
+    statusMessage->setVisible(false);
+
+    QString coverArtMessage = tr("Looking for cover art");
+    loadingProgressBar->setVisible(true);
+    loadingProgressBar->setFormat(coverArtMessage);
+    loadingProgressBar->setValue(kPercentForCoverArtLinksTask);
+
+    m_tagFetcher.startFetchCoverArtLinks(selectedTagAlbumId);
+}
+
+void DlgTagFetcher::slotCoverFound(
+        const QObject* pRequestor,
+        const CoverInfo& coverInfo,
+        const QPixmap& pixmap,
+        mixxx::cache_key_t requestedCacheKey,
+        bool coverInfoUpdated) {
+    Q_UNUSED(requestedCacheKey);
+    Q_UNUSED(coverInfoUpdated);
+    if (pRequestor == this &&
+            m_track &&
+            m_track->getLocation() == coverInfo.trackLocation) {
+        m_trackRecord.setCoverInfo(coverInfo);
+        m_pWCurrentCoverArtLabel->setCoverArt(coverInfo, pixmap);
+    }
+}
+
+void DlgTagFetcher::slotUpdateStatusMessage(const QString& message) {
+    loadingProgressBar->setVisible(false);
+    statusMessage->setVisible(true);
+    statusMessage->setText(message);
+}
+
+void DlgTagFetcher::slotStartFetchCoverArt(const QList<QString>& allUrls) {
+    int fetchedCoverArtQualityConfigValue =
+            m_pConfig->getValue(mixxx::library::prefs::kCoverArtFetcherQualityConfigKey,
+                    static_cast<int>(DlgPrefLibrary::CoverArtFetcherQuality::Low));
+
+    DlgPrefLibrary::CoverArtFetcherQuality fetcherQuality =
+            static_cast<DlgPrefLibrary::CoverArtFetcherQuality>(
+                    fetchedCoverArtQualityConfigValue);
+
+    // Cover art links task can retrieve us variable number of links with different cover art sizes
+    // Every single successful response has 2 links.
+    // These links are cover arts with 250 PX and 500 PX
+    // Some of the tags might have link of a cover art with 1200 PX
+    // Some of the tags might have link of a cover art more than 1200 PX
+    // At final, we always retrieve 2 links, but for some tags this can be 3 or 4
+    // We need to pick the correct size according to user choice
+
+    // User choices and the possible fetched cover art sizes are:
+    // Highest -> More than 1200 PX, if not available 1200 PX or 500 PX
+    // High    -> 1200 PX if not available 500 PX
+    // Medium  -> Always 500 PX fetched
+    // Low     -> Always 250 PX fetched
+
+    if (fetcherQuality == DlgPrefLibrary::CoverArtFetcherQuality::Highest) {
+        getCoverArt(allUrls.last());
+        return;
+    } else if (fetcherQuality == DlgPrefLibrary::CoverArtFetcherQuality::High) {
+        allUrls.size() < 3 ? getCoverArt(allUrls.last()) : getCoverArt(allUrls.at(2));
+        return;
+    } else if (fetcherQuality == DlgPrefLibrary::CoverArtFetcherQuality::Medium) {
+        getCoverArt(allUrls.at(1));
```

Comment:
A check for 1 is missing

Reply:
Normally all the responses at least have 2 cover art's URL, I haven't seen any response with one cover art URL.

So for all of the checks above would it be better to check if the allUrls is not empty and then the if block would do the rest? Will it work? What do you think?